src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Cmdlet/UpdateAzureSqlManagedInstance.cs

```diff
@@ -0,0 +1,241 @@
+        public int? VCore { get; set; }
+
+        /// <summary>
+        /// The tags to associate with the instance.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The tags to associate with the instance.")]
+        [Alias("Tags")]
+        public Hashtable Tag { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether or not to assign identity for instance
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Generate and assign an Azure Active Directory Identity for this instance for use with key management services like Azure KeyVault.")]
+        public SwitchParameter AssignIdentity { get; set; }
+
+        /// <summary>
+        /// Defines whether it is ok to skip the requesting of rule removal confirmation
+        /// </summary>
+        [Parameter(HelpMessage = "Skip confirmation message for performing the action")]
+        public SwitchParameter Force { get; set; }
+
+        /// <summary>
+        /// Get the instance to update
+        /// </summary>
+        /// <returns>The instance being updated</returns>
+        protected override IEnumerable<Model.AzureSqlManagedInstanceModel> GetEntity()
+        {
+            return new List<Model.AzureSqlManagedInstanceModel>() { ModelAdapter.GetManagedInstance(this.ResourceGroupName, this.Name) };
+        }
+
+        /// <summary>
+        /// Constructs the model to send to the update API
+        /// </summary>
+        /// <param name="model">The result of the get operation</param>
+        /// <returns>The model to send to the update</returns>
+        protected override IEnumerable<Model.AzureSqlManagedInstanceModel> ApplyUserInputToModel(IEnumerable<Model.AzureSqlManagedInstanceModel> model)
+        {
+            AzureSqlManagedInstanceModel existingInstance = ModelAdapter.GetManagedInstance(this.ResourceGroupName, this.Name);
+            Management.Internal.Resources.Models.Sku Sku = new Management.Internal.Resources.Models.Sku();
+
+            if (Edition != null)
+            {
+                string computeGeneration = existingInstance.Sku.Name.Contains(Constants.ComputeGenerationGen4) ? Constants.ComputeGenerationGen4 : Constants.ComputeGenerationGen5;
+                string editionShort = Edition.Equals(Constants.GeneralPurposeEdition) ? "GP" : Edition.Equals(Constants.BusinessCriticalEdition) ? "BC" : "Unknown";
+                Sku.Name = editionShort + "_" + computeGeneration;
+                Sku.Tier = Edition;
+            }
+            else
+            {
+                Sku = null;
+            }
+
+            // Construct a new entity so we only send the relevant data to the Managed instance
+            List<Model.AzureSqlManagedInstanceModel> updateData = new List<Model.AzureSqlManagedInstanceModel>();
+            updateData.Add(new Model.AzureSqlManagedInstanceModel()
+            {
+                ResourceGroupName = this.ResourceGroupName,
```

Comment:
What is the different between Set and Update? It looks like it overrides all properties in both.

Reply:
Set calls CreateOrUpdate, so if instance does not exist it will create one, and otherwise update parameters that are specified, and the rest ones will be set to default. Update just updates specified parameters. In contrast to "New", "Set" doesn't check if instance exists, but does CreateOrUpdate. I personally don't see much use for Set cmdlet, and it can potentially confuse customers, so I am fine with removing Set, if you agree? I previously added Set to Mark's suggestion, not sure if we are still pushing for both Update and Set pair of cmdlets.

Reply:
SQL commands all use Set- for updating, not Update-. Why do we want instance cmdlets to be different?

Reply:
Either is fine, I just think we should not have both. @petrajkogit Thoughts?

Reply:
I am also fine with both options. However, since logical server has 'Set', I will for consistency use 'Set' also, and remove 'Update'